chemprop/cli/utils/parsing.py

```diff
@@ -199,28 +200,83 @@ def make_datapoints(
     else:
         N = len(smiss[0])
 
+    if len(smiss) > 0:
+        molss = [[make_mol(smi, keep_h, add_h) for smi in smis] for smis in smiss]
+    if len(rxnss) > 0:
+        rctss = [
+            [
+                make_mol(f"{rct_smi}.{agt_smi}" if agt_smi else rct_smi, keep_h, add_h)
+                for rct_smi, agt_smi, _ in (rxn.split(">") for rxn in rxns)
+            ]
+            for rxns in rxnss
+        ]
+        pdtss = [
+            [make_mol(pdt_smi, keep_h, add_h) for _, _, pdt_smi in (rxn.split(">") for rxn in rxns)]
+            for rxns in rxnss
+        ]
+
     weights = np.ones(N, dtype=np.single) if weights is None else weights
     gt_mask = [None] * N if gt_mask is None else gt_mask
     lt_mask = [None] * N if lt_mask is None else lt_mask
 
     n_mols = len(smiss) if smiss else 0
-    X_d = [None] * N if X_d is None else X_d
     V_fss = [[None] * N] * n_mols if V_fss is None else V_fss
     E_fss = [[None] * N] * n_mols if E_fss is None else E_fss
     V_dss = [[None] * N] * n_mols if V_dss is None else V_dss
 
+    if X_d is None and molecule_featurizers is None:
+        X_d = [None] * N
+    else:
+        if molecule_featurizers is None:
+            pass
+        else:
+            # TODO: MorganFeaturizers take radius, length, and include_chirality as arguements.
+            # Should we expose these through the CLI?
+            molecule_featurizers = [
+                Factory.build(MoleculeFeaturizerRegistry[mf]) for mf in molecule_featurizers
+            ]
+            if len(smiss) > 0:
+                mol_descriptors = np.hstack(
+                    [
+                        np.vstack(
+                            [np.hstack([mf(mol) for mf in molecule_featurizers]) for mol in mols]
```

Comment:
When `mol.GetNumHeavyAtoms() == 0`, it should be `np.zeros(sum(len(mf) for mf in molecule_featurizers)`.

Reply:
I don't think we need to handle the case of 0 heavy atoms separately. The morgan fingerprint featurizers we currently have work when there are only hydrogens in the graph. And as we add more featurizers, I think it is the job of the featurizer to guarantee it works when there are no heavy atoms. If you disagree, can you give me an example to test?

Reply:
That is a fair point. I mentioned this because it seems that not all features are equally long, based on this line in v1. (https://github.com/chemprop/chemprop/blob/master/chemprop/data/data.py#L141)